pom.xml

```diff
@@ -1482,5 +1447,121 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>spotbugs-on-java8+</id>
+      <activation>
+        <jdk>[1.8,)</jdk>
+      </activation>
+      <dependencyManagement>
+        <dependencies>
+          <dependency>
+            <groupId>com.github.spotbugs</groupId>
+            <artifactId>spotbugs-annotations</artifactId>
+            <version>3.1.6</version>
+          </dependency>
+        </dependencies>
+      </dependencyManagement>
+
+      <dependencies>
+        <dependency>
+          <groupId>com.github.spotbugs</groupId>
+          <artifactId>spotbugs-annotations</artifactId>
+          <scope>provided</scope>
+          <optional>true</optional>
+        </dependency>
+      </dependencies>
+
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>com.github.spotbugs</groupId>
+            <artifactId>spotbugs-maven-plugin</artifactId>
+            <version>3.1.6</version>
+            <configuration>
+              <failOnError>${findbugs.failOnError}</failOnError>
+            </configuration>
+            <executions>
+              <execution>
+                <id>spotbugs</id>
+                <goals>
+                  <goal>check</goal>
+                </goals>
+                <phase>verify</phase>
+                <configuration>
+                  <!--  Do not define excludeFilterFile here as it will force a plugin to provide a file -->
+                  <!--  Instead we configure this in a profile -->
+                  <xmlOutput>true</xmlOutput>
+                  <spotbugsXmlOutput>false</spotbugsXmlOutput>
+                  <effort>${findbugs.effort}</effort>
+                  <threshold>${findbugs.threshold}</threshold>
+                </configuration>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <profile>
+      <id>findbugs-on-java7</id>
+      <activation>
+        <jdk>(,1.7]</jdk>
```

Comment:
it needs JDK 7 to run so this should be `[1.7]` should it not?

Reply:
I don't follow.   
As is if you used java 6 to build (who knows) you will activate Findbugs.  But Findbugs requires java 7 to run.  so you will activate findbugs and then it will blow up.

I am just saying the activation should be what findbugs requires (java 1.7+) and without the spotbugs activation (java 1.8 or above)  which leaves 1.7 and only 1.7...  which is `[1.7]`

Reply:
I am lost. If you set Java.level to 7 and run with jdk 8 how did what you have above help?  The jdk would activate the spotbugs profile irrespective of the java.level.

Also spotbugs can analyze code from have 1.0 up. Ref 
https://spotbugs.github.io

Reply:
I'll try once more.

As a developer I am using JDK 8 when building a maven project.
My project is however targeting Java 7.
Now the profile that is active is spotbugs because maven detects I am using JDK 8.
spotbugs does not support targeting Java 7 and will fail the build.

Now I must use JDK 7 to target Java 7 if we use maven profile activation based on JDK target. To make the build pass.

However our Jenkinsfile and ci.jenkins.io only supports building with JDK 8.
So how do I activate Findbugs when I have `java.level=7` as in: we are targeting Java 7 when I am building with JDK 8?

Most developers use JDK 8 to build Jenkins plugins so this needs to be supported.

At moment it seems like bumping major version is the only way and completely abandoning the support for building with `java.level=7`

Unless I am missing some fancy maven profile switching.

Reply:
if spotbugs can analyze code from 1.0 and up then why is the integration test failing?

Reply:
they fail because of enforcer demands compliance on JDK level 😔

```
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (display-info) @ not-newest-java-level ---
[INFO] Restricted to JDK 1.7 yet com.github.spotbugs:spotbugs-annotations:jar:3.1.6:provided contains edu/umd/cs/findbugs/annotations/NonNull.class targeted to JDK 1.8
[WARNING] Rule 2: org.apache.maven.plugins.enforcer.EnforceBytecodeVersion failed with message:
Found Banned Dependency: com.github.spotbugs:spotbugs-annotations:jar:3.1.6
Use 'mvn dependency:tree' to locate the source of the banned dependencies.
```